### ISSUE TITLE
Unable to find an entry point for certain new methods in NET

### ISSUE THREAD

**Issue by xjjius** [openbabel/openbabel#2068]
In the 3.0.0 version of Openbabel, I found some changes in the name of certain methods. For examples, the OBAtom::GetHeteroValence() in 2.4.1 was replaced by OBAtom::GetHeteroDegree() (I guess?) and OBAtom::GetHvyValence() is now GetHvyDegree(). I believe there should be other changes like this.

However, when I replaced these name in my NET code, I got this exception:

System.EntryPointNotFoundException:“Unable to find an entry point named 'CSharp_OpenBabel_OBAtom_GetHeteroDegree' in DLL 'openbabel_csharp'.”

I have also tried the same method in java, and it worked fine. So I guess this bug may be related to the NET binding.

**Comment by baoilleach** [comment]
Are you certain you are using the openbabel_csharp.dll provided in the 3.0 release? It's hard for me to explain this error otherwise.

**Comment by xjjius** [comment]
Yes. I imported OBDotNet.dll to my NET project from openbabel-3.0.0 directory. The C# code was built successfully. But l got the the error message above when I try to run it. In the 2.4.1 version, the same code runs perfectly..

I have also tried the the Java binding using the same method in openbabel-3.0.0. It worked will. 

A simple test code:

class Program
    {
        static void Main(string[] args)
        {
            string molFile = "N(CNC1)CC1";
            OBConversion conv = new OBConversion();
            conv.SetInFormat("smi");
            OBMol mol = new OBMol();
            conv.ReadString(mol, molFile);
            int numberAtoms = (int)mol.NumAtoms();

            for (int i = 1; i <= numberAtoms; i++)
            {
                OBAtom atom = mol.GetAtom(i);
                string label = atom.GetAtomType();
                uint hetero = atom.GetHeteroDegree();  // new method name for GetHeteroValence()

                Console.WriteLine($"{label} : {hetero}");
            }

            Console.ReadLine();
        }
    }

**Comment by ghutchis** [comment]
Since I don't do C#, I can't check the bindings easily. (I'm working on that.)

But my suggestion would be to install SWIG, and build the bindings from source.

**Comment by xjjius** [comment]
I think I found the problem. I installed two version of Openbabel in my computer. I had change the value of $BABEL_DATADIR to new version, but I forgot to change the value in $Path. The OBDotNet.dll was trying to load the openbabel_csharp.dll from the 2.4.1 directory which still calls the method using the old name. The problem was solved by changing the openbabel related value in $Path. Thanks for the help !!
